Tests/HummingbirdRouterTests/ControllerTests.swift

```diff
@@ -0,0 +1,148 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2021-2023 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Hummingbird
+import HummingbirdRouter
+import HummingbirdTesting
+import XCTest
+
+final class ControllerTests: XCTestCase {
+    func testRouterControllerWithSingleRoute() async throws {
+        struct TestController: RouterController {
+            typealias Context = BasicRouterRequestContext
+            var body: some RouterMiddleware<Context> {
+                Get("foo") { _,_ in "foo" }
+            }
+        }
+
+        let router = RouterBuilder(context: BasicRouterRequestContext.self) {
+            TestController()
+        }
+
+        let app = Application(responder: router)
+        try await app.test(.router) { client in
+            try await client.execute(uri: "/foo", method: .get) {
+                XCTAssertEqual(String(buffer: $0.body), "foo")
+            }
+        }
+    }
+
+    func testRouterControllerWithMultipleRoutes() async throws {
+        struct TestController: RouterController {
+            typealias Context = BasicRouterRequestContext
+            var body: some RouterMiddleware<Context> {
+                Get("foo") { _,_ in "foo" }
+                Get("bar") { _,_ in "bar" }
+            }
+        }
+
+        let router = RouterBuilder(context: BasicRouterRequestContext.self) {
+            TestController()
+        }
+
+        let app = Application(responder: router)
+        try await app.test(.router) { client in
+            try await client.execute(uri: "/foo", method: .get) {
+                XCTAssertEqual(String(buffer: $0.body), "foo")
+            }
+
+            try await client.execute(uri: "/bar", method: .get) {
+                XCTAssertEqual(String(buffer: $0.body), "bar")
+            }
+        }
+    }
+
+    func testRouterControllerWithGenericChildren() async throws {
+        struct ChildController: RouterController {
+            typealias Context = BasicRouterRequestContext
+            let name: String
+            var body: some RouterMiddleware<Context> {
+                Get("child_\(name)") { _,_ in "child_\(name)" }
+            }
+        }
+
+        struct ParentController<Context: RouterRequestContext, Child: RouterMiddleware>: RouterController where Child.Context == Context {
+            @MiddlewareFixedTypeBuilder<Request, Response, Context> var child: () -> Child
```

Comment:
Swift 5.9 doesn't seem to like this. Perhaps instead of storing the closure you store the result of the closure it might work.

Reply:
Ahh, yea I'm running on latest Xcode with 6.0. I'll check compilation on 5.9.

Reply:
I didn't have any issues compiling and running using this command...

`swift test -Xswiftc -swift-version -Xswiftc 5`

Reply:
It is Swift 5.9 compiler that is the issue, not the Swift 6 compiler in Swift 5 mode.